features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java

```diff
@@ -540,6 +600,120 @@ private BmpGlobalIpRib buildGlobalIpRib(PrefixByAS prefixByAS) {
 
     }
 
+    private BmpAsnInfo fetchAndBuildAsnInfo(Long asn) {
+        Optional<AsnInfo> asnInfoOptional = BmpWhoIsClient.getAsnInfo(asn);
+        if (asnInfoOptional.isPresent()) {
+            BmpAsnInfo bmpAsnInfo = new BmpAsnInfo();
+            AsnInfo asnInfo = asnInfoOptional.get();
+            bmpAsnInfo.setAsn(asnInfo.getAsn());
+            bmpAsnInfo.setOrgId(asnInfo.getOrgId());
+            bmpAsnInfo.setAsName(asnInfo.getAsName());
+            bmpAsnInfo.setOrgName(asnInfo.getOrgName());
+            bmpAsnInfo.setAddress(asnInfo.getAddress());
+            bmpAsnInfo.setCity(asnInfo.getCity());
+            bmpAsnInfo.setStateProv(asnInfo.getStateProv());
+            bmpAsnInfo.setPostalCode(asnInfo.getPostalCode());
+            bmpAsnInfo.setCountry(asnInfo.getCountry());
+            bmpAsnInfo.setSource(asnInfo.getSource());
+            bmpAsnInfo.setRawOutput(asnInfo.getRawOutput());
+            bmpAsnInfo.setRemarks(asnInfo.getRemarks());
+            bmpAsnInfo.setLastUpdated(Date.from(Instant.now()));
+            return bmpAsnInfo;
+        }
+        return null;
+    }
+
+    private BmpRouteInfo fetchAndBuildRouteInfo(String prefix) {
+        Optional<RouteInfo> routeInfoOptional = BmpWhoIsClient.getRouteInfo(prefix);
+        if (routeInfoOptional.isPresent() && routeInfoOptional.get().getPrefix() != null) {
+            RouteInfo routeInfo = routeInfoOptional.get();
+            Integer prefixLen = routeInfo.getPrefixLen();
+            Long originAs = routeInfo.getOriginAs();
+            BmpRouteInfo bmpRouteInfo = bmpRouteInfoDao.findByPrefix(routeInfo.getPrefix(), prefixLen, originAs);
+            if (bmpRouteInfo == null) {
+                bmpRouteInfo = new BmpRouteInfo();
+                bmpRouteInfo.setPrefix(routeInfo.getPrefix());
+                bmpRouteInfo.setPrefixLen(routeInfo.getPrefixLen());
+                bmpRouteInfo.setDescr(routeInfo.getDescription());
+                bmpRouteInfo.setOriginAs(routeInfo.getOriginAs());
+                bmpRouteInfo.setSource(routeInfo.getSource());
+            }
+            bmpRouteInfo.setLastUpdated(Date.from(Instant.now()));
+            return bmpRouteInfo;
+        }
+        return null;
+    }
+
+    private List<BmpAsnPathAnalysis> buildBmpAsnPath(String asnPath) {
+
+        List<BmpAsnPathAnalysis> bmpAsnPathAnalyses = new ArrayList<>();
+        String[] asnArray = asnPath.split(" ");
```

Comment:
Why not parse the ASNs into `long`s here? this would save some error handling in the loop.
By doing so, we can skip the whole update if the path in invalid.